zarr/cupy.py

```diff
@@ -0,0 +1,59 @@
+from numcodecs.abc import Codec
+from numcodecs.registry import get_codec, register_codec
+
+from .util import ensure_contiguous_ndarray
+
+
+class CuPyCPUCompressor(Codec):
+    """CPU compressor for CuPy arrays
+
+    This compressor converts CuPy arrays host memory before compressing
+    the arrays using `compressor`.
```

Comment:
Should we be using host compressors on GPU data? Or should be using different compressors like those from [nvCOMP]( https://github.com/NVIDIA/nvcomp )?

cc @thomcom (in case you have thoughts here 🙂)

Reply:
Good point, it would make sense to implement a compressor, `CuPyGPUCompressor`,  that do the compression on the GPU.

However, in order to limit the scope of this PR, I think we should use the existing CPU compressors for now. Then in a follow up PR, we can implement compression directly in the GPU using something like nvCOMP.